src/KDSoapClient/KDSoapMessageAddressingProperties.cpp

```diff
@@ -217,16 +217,34 @@ KDSoapMessageAddressingProperties::~KDSoapMessageAddressingProperties()
 
 QString KDSoapMessageAddressingProperties::predefinedAddressToString(KDSoapMessageAddressingProperties::KDSoapAddressingPredefinedAddress address, KDSoapAddressingNamespace addressingNamespace)
 {
-    const QString addressingNS = addressingNamespaceToString(addressingNamespace);
+    QString prefix = addressingNamespaceToString(addressingNamespace);
+    // Up to and including the 2004/08 spec the well-known URIs have a /role/
+    // folder in their path and furthermore only /anonymous is actually
+    // well-known.
+    switch (addressingNamespace) {
+    case Addressing200303:
+    case Addressing200403:
+    case Addressing200408: {
+        prefix += QLatin1String("/role");
+        if (address != Anonymous) {
+            qWarning("Anything but Anonymous has no meaning in ws-addressing 2004/08");
```

Comment:
As noted in https://github.com/caspermeijn/KDSoap/pull/1:
I think that a assert would be better. This is more explicit during development. For example:
Q_ASSERT(false); // should never happen

Reply:
asserts now